pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix

```diff
@@ -0,0 +1,764 @@
+
+  });
+
+  hspec = overrideCabal super.hspec (drv: {
+    version         = "2.4.7";
+    sha256          = "1jvf7x43gkch4b8nxqdascqlh4rh2d1qvl44skwqkz0gw154ldan";
+    doHaddock       = false;
+  });
+
+  hspec-core = overrideCabal super.hspec-core (drv: {
+    version         = "2.4.7";
+    sha256          = "0syjbx3s62shwddp75qj0nfwmfjn0yflja4bh23x161xpx1g0igx";
+    doHaddock       = false;
+  });
+
+  hspec-discover = overrideCabal super.hspec-discover (drv: {
+    version         = "2.4.7";
+    sha256          = "1cgj6c6f5vpn36jg2j7v80nr87x1dsf7qyvxvjw8qimjdxrcx0ba";
+    doHaddock       = false;
+  });
+
+  hspec-meta = overrideCabal super.hspec-meta (drv: {
+    version         = "2.4.6";
+    sha256          = "0qmvk01n79j6skn79r6zalg2pd0x0nqqn9qn8mhg0pgyzcdnfc9b";
+    doHaddock       = false;
+  });
+
+  microlens = overrideCabal super.microlens (drv: {
+    version         = "0.4.8.3";
+    sha256          = "17qx2mbqdrlnkc3gxq8njbp7qw8nh51drmz6fc8khgj9bls5ni2k";
+
+  });
+
+  microlens-mtl = overrideCabal super.microlens-mtl (drv: {
+    version         = "0.1.11.1";
+    sha256          = "0l6z1gkzwcpv89bxf5vgfrjb6gq2pj7sjjc53nvi5b9alx34zryk";
+    doHaddock       = false;
+  });
+
+  microlens-th = overrideCabal super.microlens-th (drv: {
+    version         = "0.4.1.3";
+    sha256          = "15a12cqxlgbcn1n73zwrxnp2vfm8b0ma0a0sdd8zmjbs8zy3np4f";
+    doHaddock       = false;
+  });
+
+  nanospec = overrideCabal super.nanospec (drv: {
+    version         = "0.2.2";
+    sha256          = "1rcmhl9bhyfvanalnf1r86wkx6rq6wdvagnw1h011jcnnb1cq56g";
+    jailbreak       = true;
+  });
+
+  pretty-show = overrideCabal super.pretty-show (drv: {
+    version         = "1.6.16";
+    sha256          = "0l03mhbdnf0sj6kw2s3cf2xhfbl0809jr9fhj7cmpkhjpxv89vnv";
+
+  });
+
+  singletons = overrideCabal super.singletons (drv: {
+    version         = "2.4.1";
+    sha256          = "1kzrl9njvkbvxylk9jg61vy3ksmxmzymci5hdp0ilpsah4620yjx";
+    doCheck         = false;
+    jailbreak       = true;
+  });
+
+  tasty = overrideCabal super.tasty (drv: {
+    version         = "1.0.0.1";
+    sha256          = "0ggqffw9kbb6nlq1pplk131qzxndqqzqyf4s2p7576nljx11a7qf";
+
+  });
+
+  test-framework = overrideCabal super.test-framework (drv: {
+    version         = "0.8.2.0";
+    sha256          = "1hhacrzam6b8f10hyldmjw8pb7frdxh04rfg3farxcxwbnhwgbpm";
+    doCheck         = false;
+    jailbreak       = true;
+    editedCabalFile = null;
+    revision        = null;
+  });
+
+  xml-conduit = overrideCabal super.xml-conduit (drv: {
+    version         = "1.7.1.0";
+    sha256          = "1c4ip76qgqjdyf77h97mf3yxdimv7m5ma5v20wchn9qjmbkr8ffa";
+
+  });
+
+  yaml = overrideCabal super.yaml (drv: {
+    version         = "0.8.28";
+    sha256          = "0swgkzkfrwj0ac7lssn8rnrdfmh3lcsdn5fbq2iwv55di6jbc0pp";
+
+  });
+
+
+# Upstreamed
+
+  constraints = overrideCabal super.constraints (drv: {
```

Comment:
I really don't like these overrides because they unconditionally override whatever version `hackage-packages.nix` provides -- even if the version in there is actually newer than the one we hard-code here. I'd very much prefer to have these overrides import the relevant *patches*, because then we'd at least notice when they cease to apply to the upstream version.

Reply:
Completely agreed -- my plan was to submit early, and then pare down the Hackage-sourced overrides, as they're overtaken by the hackage-packages regeneration.